Add list of contributors to the README

JohanSchott opened the pull request:
Several people have contributed to the impurityModel repo.

I thought it could be nice to recognise this by making a list of the people that have contributed in some way.

I also wrote down a very short description of each persons contribution (please feel free to edit, especially your own contribution description!).

@johanjoensson, @fesorg, @patrikthunstrom, @PetterSaterskog, @kalvdans, and @igordimarco, please let me know what you think about this PR.

kalvdans commented on README.md:
```diff
@@ -138,9 +135,27 @@ make -s -C doc/sphinx html
+- Felix Sorgenfrei (@fesorg): First author in [publication](https://journals.aps.org/prb/abstract/10.1103/PhysRevB.109.115126) using the impurityModel repo. Has also made the related repo: https://github.com/fesorg/Tutorial-X-ray-from-RSPt
+- Patrik Thunström (@patrikthunstrom): Involved in discussions about computational algorithms, reported bugs, and has provided theoretical knowledge and inspiration.
+- Petter Säterskog (@kpws): Written some of the initial key functionalities.
+- Christian Häggström (@kalvdans): Has provided valueble reviews on PRs.
```
```suggestion
- Christian Häggström (@kalvdans): Has provided valuable reviews on PRs.
```

Living up to my expectations of commenting on details :D

JohanSchott commented:
Thanks @fesorg @johanjoensson and @kalvdans for reviewing and approving the PR. 

I added a commit where I corrected the github username for Petter Säterskog.
